pkg/bootstrap/crds.go

```diff
@@ -181,35 +182,52 @@ func processCRDFile(ctx context.Context, cl client.Client, fileName string) erro
 	return utilerrors.NewAggregate(allErrors)
 }
 
-func decodeCRD(ctx context.Context, raw string) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
+func decodeCRD(ctx context.Context, raw string) (*apiextensionsv1.CustomResourceDefinition, error) {
 	log := common.LogFromContext(ctx)
 	rawJSON, err := yaml.YAMLToJSON([]byte(raw))
 	if err != nil {
 		log.Error(err, "unable to convert raw data to JSON")
 		return nil, err
 	}
-	obj := &apiextensionsv1beta1.CustomResourceDefinition{}
-	_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, obj)
+	v1beta1obj := &apiextensionsv1beta1.CustomResourceDefinition{}
+	_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, v1beta1obj)
 	if err != nil {
-		log.Error(err, "unable to decode object into Unstructured")
-		return nil, err
+		v1obj := &apiextensionsv1.CustomResourceDefinition{}
+		_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawJSON, nil, v1beta1obj)
+		if err != nil {
+			log.Error(err, "unable to decode object into CustomResourceDefinition")
+			return nil, err
+		}
+		if v1obj.GroupVersionKind().GroupKind().String() == "CustomResourceDefinition.apiextensions.k8s.io" {
+			return v1obj, nil
+		}
+		return nil, nil
 	}
-	if obj.GroupVersionKind().GroupKind().String() == "CustomResourceDefinition.apiextensions.k8s.io" {
-		return obj, nil
-	} else {
+	if v1beta1obj.GroupVersionKind().GroupKind().String() != "CustomResourceDefinition.apiextensions.k8s.io" {
 		return nil, nil
```

Comment:
should return an error?

Reply:
That would be changing the original behavior.  If we return an error, then CRD processing fails as will all reconciles from this point on.  I believe all this does is silently fails to keep reconcile going, without breaking everything, as the failure is unrecoverable (i.e. no user action can fix it).  I'll add a log message though, so there's something visible.